src/main/scala/io/iohk/ethereum/ledger/Ledger.scala

```diff
@@ -317,13 +317,16 @@ class Ledger(vm: VM) extends Logger {
 
   private def saveNewContract(address: Address, result: PR, config: EvmConfig): PR = {
```

Comment:
The code looks good and i'm still running to see what block do I get. Do you think it's too much trouble to create a test to verify this if branches are ok?